pkg/api/customization/authn/user_actions.go

```diff
@@ -29,28 +29,25 @@ type Handler struct {
 }
 
 func (h *Handler) Actions(actionName string, action *types.Action, apiContext *types.APIContext) error {
-	resetFirstLogin := false
 	switch actionName {
 	case "changepassword":
-		err := h.changePassword(actionName, action, apiContext)
-		if err == nil {
-			resetFirstLogin = true
+		if err := h.changePassword(actionName, action, apiContext); err != nil {
+			return err
 		}
-		return err
 	case "setpassword":
-		err := h.setPassword(actionName, action, apiContext)
-		if err == nil {
-			resetFirstLogin = true
+		if err := h.setPassword(actionName, action, apiContext); err != nil {
+			return err
 		}
-		return err
+	default:
+		return errors.Errorf("bad action %v", actionName)
 	}
 
-	if resetFirstLogin && strings.EqualFold(settings.FirstLogin.Get(), "true") {
+	if strings.EqualFold(settings.FirstLogin.Get(), "true") {
```

Comment:
I'd probably change this to `!strings.EqualsFold(...., "false")` so that you handle empty and wrong values of `first-login`